vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -4,72 +4,101 @@
       NUMBER_OF_BMH: "{{ NUM_OF_MASTER_REPLICAS|int +  NUM_OF_WORKER_REPLICAS|int }}"
 
   - name: Update maxSurge and maxUnavailable fields
-    shell: |
-           kubectl get machinedeployment -n {{NAMESPACE}} test1 -o json | 
-           jq '.spec.strategy.rollingUpdate.maxSurge=1|.spec.strategy.rollingUpdate.maxUnavailable=1' | 
-           kubectl apply -f-
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    community.kubernetes.k8s:
+      api_version: cluster.x-k8s.io/v1alpha3
+      kind: MachineDeployment
+      name: "{{ CLUSTER_NAME }}"
+      namespace: "{{ NAMESPACE }}"
+      resource_definition:
+        spec:
+          strategy:
+            rollingUpdate:
+              maxSurge: 1
+              maxUnavailable: 1
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
```

Comment:
In general, are those scale operations working as intended? Because we have seen it silently ignoring the intended change and setting the default value for `maxSurge` field in node reuse tests which resulted in hiding the bug. 

cc @mboukhalfa

Reply:
This should work because in node_reuse I was using `apply: yes ` which ignores those fields.

Reply:
Yes scaling has worked well here. :slightly_smiling_face: